ui/raidboss/data/05-shb/raid/e12s.js

```diff
@@ -440,6 +474,109 @@ export default {
+          // to get duplicates of the statues
+          const statue = {
+            id: matches.targetId,
+          };
+          data.statues.push(statue);
+          const statueId = parseInt(matches.targetId, 16);
+          const statueData = await window.callOverlayHandler({
+            call: 'getCombatants',
+            ids: [statueId],
+          });
+
+          if (statueData === null || !statueData.combatants || !statueData.combatants.length)
+            return;
+
+          // Center = 0, -75
+          const x = statueData.combatants[0].PosX;
+          const y = statueData.combatants[0].PosY - -75;
+          statue.dist = Math.hypot(x, y);
+          if (data.statues.length > 1) {
+            // Sort so that closest statue (inner) is first
+            data.statues.sort((a, b) => a.dist - b.dist);
+          }
+        }
+      },
+      infoText: (data, matches, output) => {
+        // Player is the source of the tether for this one, oddly enough
+        if (data.me !== matches.source)
+          return;
+
+        if (data.statues.length > 0 && data.statues[0].id === matches.targetId)
+          data.statueDir = 'inner';
+        else if (data.statues.length > 1 && data.statues[1].id === matches.targetId)
+          data.statueDir = 'outer';
+        else
+          data.statueDir = 'unknown';
+
+        return output[data.statueDir]();
+      },
+      outputStrings: {
+        inner: {
+          en: 'Inner Statue',
+        },
+        outer: {
+          en: 'Outer Statue',
+        },
+        unknown: {
+          en: '???',
+        },
+      },
+    },
+    {
+      id: 'E12S Promise Palm Of Temperance SE',
+      netRegex: NetRegexes.startsUsing({ source: 'Guardian Of Eden', id: '58B4', capture: false }),
+      durationSeconds: 10,
+      infoText: (_1, _2, output) => output.knockback(),
+      outputStrings: {
+        knockback: {
+          en: 'Knockback SE',
+        },
+      },
+    },
+    {
+      id: 'E12S Promise Palm Of Temperance SW',
+      netRegex: NetRegexes.startsUsing({ source: 'Guardian Of Eden', id: '58B5', capture: false }),
+      durationSeconds: 10,
+      infoText: (_1, _2, output) => output.knockback(),
+      outputStrings: {
+        knockback: {
+          en: 'Knockback SW',
+        },
+      },
+    },
+    {
+      // Time the first dodge for lasers from the left/right split
+      id: 'E12S Promise Statue Laser First',
+      netRegex: NetRegexes.startsUsing({ source: 'Eden\'s Promise', id: '58A[DE]', capture: false }),
+      netRegexDe: NetRegexes.startsUsing({ source: 'Edens Verheißung', id: '58A[DE]', capture: false }),
+      netRegexFr: NetRegexes.startsUsing({ source: 'Promesse D\'Éden', id: '58A[DE]', capture: false }),
+      netRegexJa: NetRegexes.startsUsing({ source: 'プロミス・オブ・エデン', id: '58A[DE]', capture: false }),
+      condition: (data) => data.statueDir !== undefined && !data.statueLaserCount,
```

Comment:
If statueDir is undefined, could treat it as "unknown"?

Reply:
This trigger is based on the Rapturous Reach left/right that happens at the beginning. The check against `data.statueDir` is to ensure that the trigger only happens during the correct phase.

Technically the second half of the condition (`!data.statueLaserCount`) is superfluous in this case and can probably be removed, since the 4th laser should clean up all the statue-related data properties.

Reply:
Oh, I see!  Thanks for the explanation.  Can you leave a comment to that effect?